core/src/http.rs

```diff
@@ -106,16 +106,78 @@ impl Display for HttpErrorResponse {
 /// value. Returns `None` if there is no `authorization` header, and an error if there is an
 /// `authorization` header whose value is not a bearer token.
 pub fn extract_bearer_token(conn: &Conn) -> Result<Option<AuthenticationToken>, anyhow::Error> {
-    if let Some(authorization_value) = conn.request_headers().get("authorization") {
-        if let Some(received_token) = authorization_value.to_string().strip_prefix("Bearer ") {
-            return Ok(Some(
-                AuthenticationToken::new_bearer_token_from_string(received_token)
-                    .context("invalid bearer token")?,
-            ));
-        } else {
+    if let Some(authorization) = conn
+        .request_headers()
+        .get("authorization")
+        .map(HeaderValue::to_string)
+    {
+        let (bearer, token) = authorization
+            .split_once(char::is_whitespace)
+            .ok_or_else(|| anyhow!("invalid bearer token"))?;
+
+        if bearer.to_lowercase() != "bearer" {
```

Comment:
nit:
```suggestion
        let (auth_scheme, token) = authorization
            .split_once(char::is_whitespace)
            .ok_or_else(|| anyhow!("invalid scheme in authorization header"))?;

        if auth_scheme.to_lowercase() != "bearer" {
```


To line up with the `credentials = auth-scheme [ 1*SP ( token68 / #auth-param ) ]` in the Backaus-whatever notation.